File: 백준/Bronze/14696. 딱지놀이/딱지놀이.py
for _ in range(int(input())):
    a = list(map(int, input().split()))[1:]
    b = list(map(int, input().split()))[1:]
    if a.count(4) != b.count(4):
        print('A' if a.count(4) > b.count(4) else 'B')
    elif a.count(3) != b.count(3):
        print('A' if a.count(3) > b.count(3) else 'B')
    elif a.count(2) != b.count(2):
        print('A' if a.count(2) > b.count(2) else 'B')
    elif a.count(1) != b.count(1):
        print('A' if a.count(1) > b.count(1) else 'B')
    else:
        print('D')

File: 백준/Bronze/14696. 딱지놀이/딱지놀이.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : 'run/input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const n = parseInt(input[0]);

for (let i = 0; i < n; i++) {
  const a = input[i * 2 + 1].split(' ').map(Number).slice(1);
  const b = input[i * 2 + 2].split(' ').map(Number).slice(1);

  const countA = Array(5).fill(0);
  const countB = Array(5).fill(0);

  a.forEach((n) => countA[n]++);
  b.forEach((n) => countB[n]++);

  let result = 'D';

  for (let i = 4; i >= 1; i--) {
    if (countA[i] !== countB[i]) {
      result = countA[i] > countB[i] ? 'A' : 'B';
      break;
    }
  }

  console.log(result);
}